### DISCUSSION TITLE
long term plan for license?

### DISCUSSION THREAD

**Discussion by tomdavidson** [moonrepo/moon#346] [Q&A] [answered]
What is the long term licensing plan (v1, v2 etc)? Currently moonrepo/moon is MIT as well as moonrepo/dev, but the latter has a different config for OSS in .moon/oss . Im excited about moon and what I understand of it so far, just want to invest in an opensource future and hope to avoid something that might "pull an Elastic".

Will moon always be available with a libre OSI approved opensource license ? 

Are there plans for hybrid licensing - OSI but with such ans such clause or "open core" ? What are the license holder's opinion on "open core"?

**Comment by milesj** [comment, marked as answer]
@tomdavidson The moon tool (this repo) will always be free and available for public use.

In the future, there will be additional tools/features/etc that will be provided that will not be free, but those will be opt-in and not critical to functioning.

**Reply by tomdavidson** [reply]
@milesj thanks for addressing my questions so quickly. Sounds a bit like open core, but there will not be a mixing of proprietary and open source in this moonrepo/moon repo.  Just gauging the spectrum of open core here, but can 3rd parties provide tools and features on the same class? moon doesnt know of the plugins at build time kind of thing?

**Reply by milesj** [reply]
Yeah it can be seen like the open core model. The core functionality will be free, but additional features like remote caching will not be, since that requires upstream bandwidth/storage costs. The features that are not free will also not be critical for day-to-day development, and are mostly for scaling for large companies.

As for third-party, I would love a plugin system for moon, so that consumers can write their own integrations, but plugins are _very difficult_ in Rust, maybe impossible at this time. I'm hoping that rome, swc, or some of the other Rust based tools will solve this problem in the future.

**Reply by tomdavidson** [reply]
Thanks again @milesj  - for your answers and for moon.

**Reply by milesj** [reply]
@tomdavidson Of course. If you have other questions, feel free.
